api/api.cc

```diff
@@ -233,5 +233,32 @@ future<> set_server_done(http_context& ctx) {
     });
 }
 
+void req_params::process(const request& req) {
+    // Process mandatory parameters
+    for (auto& [name, ent] : params) {
```

Comment:
Just a reminder, path parameters cannot be optional or missing, or the handler wouldn't have been called:
for example, if you register a handler on: `/my-path/{keyspace}`
if you call `/my-path` the handler will not be called

And though currently not supported, but theoretically can be in the future, query paramters can repeat multiple times

Reply:
> Just a reminder, path parameters cannot be optional or missing, or the handler wouldn't have been called: for example, if you register a handler on: `/my-path/{keyspace}` if you call `/my-path` the handler will not be called

True, but it doesn't hurt to check this again here, and verify the completeness of the api implementation, to make sure it parses all the mandatory parameters.

> And though currently not supported, but theoretically can be in the future, query paramters can repeat multiple times

Actually, the mechanism here is a good basis for supporting repeated params.
We can easily collect them into a list, and even automatically parse comma-separated values to generate a list of sstrings instead of a single value that contains the commas.